.travis.yml

```diff
@@ -1,5 +1,21 @@
 language: haskell
 
+ghc: 7.8
+
+addons:
+  apt:
+    packages:
+    - ocaml
+    - camlidl
+
+sudo: false
+
+cache:
+  apt: true
+  directories:
```

Comment:
I know [Halcyon](https://halcyon.sh/) handles this well, but I don't know about Cabal. It's hacky, but the Travis script could throw away the cache and reinstall if Cabal says there will be a conflict.